ext/fg/js/float.js

```diff
@@ -193,4 +161,45 @@ class DisplayFloat extends Display {
             return '';
         }
     }
+
+    _logMessageError(event, type) {
+        yomichan.logWarning(new Error(`Popup received invalid message from origin ${JSON.stringify(event.origin)}: ${type}`));
+    }
+
+    _initialize(params) {
+        if (this._token !== null) { return; } // Already initialized
+        if (!isObject(params)) { return; } // Invalid data
+
+        const secret = params.secret;
+        if (secret !== this._secret) { return; } // Invalid authentication
+
+        const {token, frameId} = params;
+        this._token = token;
+
+        apiSendMessageToFrame(frameId, 'popupInitialized', {secret, token});
+    }
+
+    async _configure({messageId, frameId, popupId, optionsContext, childrenSupported, scale}) {
+        this.optionsContext = optionsContext;
+
+        await this.updateOptions();
+
+        if (childrenSupported && !this._initializedNestedPopups) {
+            const {depth, url} = optionsContext;
+            popupNestedInitialize(popupId, depth, frameId, url);
+            this._initializedNestedPopups = true;
+        }
+
+        this.setContentScale(scale);
+
+        apiSendMessageToFrame(frameId, 'popupConfigured', {messageId});
+    }
```

Comment:
Because the `_initializedNestedPopups` guard was moved after `await`, the function can be called multiple times. None of the lines except the block initializing nested popups seem to misbehave when called multiple times, but it could be more clear to rename the guard and put it to the beginning.

Reply:
It was an intentional choice. I didn't see any reason to prevent "configure" from being called more than once, but there was an explicit reason to prevent popup initialization multiple times. My thought was that it acts sort of like a multi-setter/`setVariousOptionsAtOnce` type function. In practice, it's currently not called more than once, but it could be in case it ever becomes necessary.